src/main/scala/mesosphere/marathon/api/akkahttp/v2/LeaderController.scala

```diff
@@ -0,0 +1,75 @@
+package mesosphere.marathon
+package api.akkahttp.v2
+
+import akka.http.scaladsl.server.Route
+import com.typesafe.scalalogging.StrictLogging
+import com.wix.accord.{ Failure, Success }
+import mesosphere.marathon.api.v2.Validation
+import mesosphere.marathon.api.akkahttp.{ Controller, Rejections }
+import mesosphere.marathon.core.election.ElectionService
+import mesosphere.marathon.plugin.auth._
+import mesosphere.marathon.raml.RuntimeConfiguration
+import mesosphere.marathon.storage.repository.RuntimeConfigurationRepository
+import mesosphere.marathon.stream.UriIO
+
+import scala.async.Async._
+import scala.concurrent.ExecutionContext
+
+case class LeaderController(
+    val electionService: ElectionService,
+    val runtimeConfigRepo: RuntimeConfigurationRepository)(
+    implicit
+    val authenticator: Authenticator,
+    val authorizer: Authorizer,
+    val executionContext: ExecutionContext) extends Controller with Validation with StrictLogging {
+
+  import mesosphere.marathon.api.akkahttp.Directives._
+  import mesosphere.marathon.api.akkahttp.EntityMarshallers._
+
+  def leaderInfo(): Route =
+    authenticated.apply { implicit identity =>
```

Comment:
Can we move `authenticated.apply { implicit identity =>` between the lines 63 and 64?

Reply:
Hm, `authorized` requires the identity but `authorized` cannot be moved since the resources diff (`UpdateResource` vs `ViewResource`).